pulse2percept/stimuli.py

```diff
@@ -173,81 +275,75 @@ def image2pulsetrain(img, implant, coding='amplitude', valrange=[0, 50],
     pulsetype : {'cathodicfirst', 'anodicfirst'}, optional
         A cathodic-first pulse has the negative phase first, whereas an
         anodic-first pulse has the positive phase first.
+
+    Returns
+    -------
+    pulses : list
+        A list of p2p.stimuli.PulseTrain objects, one for each electrode in
+        the implant.
+
     """
-    try:
-        from skimage.io import imread
-        from skimage.transform import resize
-        from skimage.color import rgb2gray
-    except ImportError:
-        raise ImportError("You do not have scikit-image installed.")
+    if not has_skimage:
+        # We don't want to repeatedly import Scikit-Image. This would (e.g.)
+        # unnecessarily slow down `video2pulsetrain`.
+        raise ImportError("You do not have scikit-image installed. "
+                          "You can install it via $ pip install scikit-image.")
 
     # Make sure range of values is valid
     assert len(valrange) == 2 and valrange[1] > valrange[0]
 
-    if not isinstance(implant, implants.ElectrodeArray):
-        raise TypeError("implant must be of type implants.ElectrodeArray.")
+    isargus1 = isinstance(implant, implants.ArgusI)
```

Comment:
Maybe keep an `allowed_implants` variable around (as a module-wide constant?), and check that the implant is one of them?